app/views/hosts/_assign_hosts.html.erb

```diff
@@ -1,6 +1,6 @@
 <%= javascript "jquery.cookie", "host_checkbox" %>
 <% assign_selected_hosts_taxonomy_path = (@taxonomy_type == "Location") ? assign_selected_hosts_location_path : assign_selected_hosts_organization_path %>
-<%= form_for taxonomy, :url => assign_selected_hosts_taxonomy_path do |f| %>
+<%= form_for taxonomy, :url => assign_selected_hosts_taxonomy_path, :method => 'PUT' do |f| %>
```

Comment:
you can probably fix it for most forms, as we already override the form_for helper, see https://github.com/theforeman/foreman/blob/develop/app/helpers/layout_helper.rb#L385